astropy/cosmology/core.py

```diff
@@ -777,6 +791,46 @@ def nu_relative_density(self, z):
 
         return prefac * self._neff_per_nu * rel_mass
 
+    def _get_nufunc_scalar(self):
+        """ Return the version of the nu relative density function
+        to use in integrals."""
+
+        if self._massivenu:
+            return self._massivenu_relative_density_scalar
+        else:
+            return lambda x : 0.22710731766 * self._Neff
```

Comment:
I'm trying to avoid cython as hard as I can.  Even ugly python code is easier to maintain than cython.

But somebody should actually try it, see what it buys us.  Just maybe in another pull!  

If nothing else, this one will give us something meaningful to compare the cython performance to.